enterprise/internal/scim/mock_db.go

```diff
@@ -44,7 +38,11 @@ func getMockDB() *database.MockDB {
 	})
 	userStore.CountFunc.SetDefaultReturn(4, nil)
 	userStore.CreateFunc.SetDefaultHook(func(ctx context.Context, user database.NewUser) (*types.User, error) {
-		newUser := types.UserForSCIM{User: types.User{ID: 5, Username: user.Username, DisplayName: user.DisplayName}}
+		nextID := 0
+		if len(users) > 0 {
+			nextID = int(users[len(users)-1].ID) + 1
+		}
```

Comment:
The `nextID` should be `1` if there are no users given, since database starts numbering at `1` and userID with `0` is considered invalid or internal (depending on the use case).

Nit: Theoretically, we should pick max + 1, since there's no guarantee that the data is ordered by ID. 

```suggestion
		maxID := 0
		for _, user := range users {
			if user.ID > maxID {
				maxID = user.ID
			}
		}
		nextID := maxID + 1
```

Reply:
Ah, nice catch! Also, I'll add a comment that user IDs should be ascending when passed to `getMockDB`.